src/Symfony/Component/HttpClient/ServerSentEvents/EventSource.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Symfony\Component\HttpClient\ServerSentEvents;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+final class EventSource implements EventSourceInterface
+{
+    private $url;
+    private $client;
+    private $reconnectionTime = 5;
+    private $shouldClose = false;
+    private $lastEventId = null;
+    private $response = null;
+
+    const CONNECTING = 0;
+    const OPEN = 1;
+    const CLOSED = 2;
+
+    public $readyState = self::CLOSED;
+
+    public function __construct(string $url, array $defaultOptions = [], HttpClientInterface $client = null)
```

Comment:
This signature prevents using the class as a service, isn't it?
Unless we expect ppl to configure one service per URL?